gradio/components.py

```diff
@@ -335,6 +338,19 @@ def base64_to_temp_file_if_needed(
         self.temp_files.add(full_temp_file_path)
         return full_temp_file_path
 
+    def pil_to_temp_file(self, img: _Image.Image, dir: str) -> str:
+        filename = str(Path(dir) / f"{self.hash_bytes(img.tobytes())}.png")
```

Comment:
Nit: Note that this method creates files with the following filepath:

`<temp_dir>/<hash>.png`

whereas the other methods create filepaths like this:

`<temp_dir>/<hash>/name.suffix`

Maybe it's better to follow the same syntax and do something like 

`<temp_dir>/<hash>/image.png`

because otherwise the filenames are very long when you download the files.

Reply:
Done!

Reply:
This isn't a breaking change but now users will get `audio (13).wav` as opposed to a unique name in their downloads dir